File: Hot Potato/hot_potato.cpp
#include <iostream>
#include <string.h>
#include <queue>

using namespace std;

string hot_potato(queue<string> names, int repetitions) {
    for (int i = 0; i < repetitions; i++) {
        string person = names.front();
        names.pop();
        names.push(person);
    }

    string personWhoGotCaught = names.front();
    names.pop();
    return personWhoGotCaught;
}

int main() {
    queue<string> names;
    names.push("Nick");
    names.push("John");
    names.push("Connor");
    names.push("Mike");
    names.push("Paul");

    cout << hot_potato(names, 12);

    return 0;
}

File: Hot Potato/hot_potato.java
import java.util.LinkedList;
import java.util.Queue;

public class hot_potato {

    public static void main(String[] args){
        Queue<String> names = new LinkedList<>();

        names.add("Nick");
        names.add("John");
        names.add("Connor");
        names.add("Mike");
        names.add("Paul");

        System.out.println(hotPotato(names, 12));

    }

    public static String hotPotato(Queue<String> names, int repetitions){
        for (int i=0; i<repetitions; i++){
            String person = names.poll();
            names.add(person);
        }
        String personWhoGotCaught = names.poll();

        return personWhoGotCaught;
    }
}
